### PR TITLE
posix: add getdents einval and enoent handling

### PR DESCRIPTION
**Pull request by llugin**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/366)

### CODE REVIEW

File: src/libpmemfile-posix/getdents.c [hunk 1]
```diff
@@ -288,6 +292,10 @@ pmemfile_getdents_worker(PMEMfilepool *pfp, PMEMfile *file, char *data,
 		++file->offset;
 	}
 
+	if (read1 == 0) {
+		return -ENOENT;
+	}
```
Comment (marcinslusarz) [inline src/libpmemfile-posix/getdents.c]: Please remove curly braces for single line statements.

**Comment by llugin** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/libpmemfile-posix/getdents.c, line 297 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/366#-KuTaPdsVdv2NUHKrBhL-r1-297:-KuTxo_C8aCRERunORdK:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/855721c299bee13c6dbd8ace94b757e4c7a72747/src/libpmemfile-posix/getdents.c#L297)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Please remove curly braces for single line statements.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/366)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/366#-:-KuUA27KtrUOaLA3WQyc:bnfp4nl)*

**Comment by sarahjelinek** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/366#-:-KuUSkISPGbnMuVqpVIa:bnfp4nl)*

**Pull request merged**